### PR TITLE
Add date utility functions

### LINKED ISSUE #9: Add date utils
I would start adding some simple date utility functions like:
- IsValid
- IsEndOfMonth
- DifferenceInDays

For string conversions (parse/format) it might be easier to wrap some existing library

### PR DESCRIPTION
**Pull request by drebrez**
Resolves #9

### CODE REVIEW

File: src/lib/date.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+export function dateIsValid(date: Date): boolean {
+  return isDate(date) && isValid(date);
+}
+
+/**
+ * Check whether the date is the end of the month
+ * @param date The date
+ * @returns A boolean indicating whether the date is the end of the month
+ */
+export function dateIsEndOfMonth(date: Date): boolean {
```
Comment (neotrow) [inline src/lib/date.ts:17]: end of month is a bit vague, is last day of month would be more precise.

**Review by neotrow — CHANGES_REQUESTED** [review]

File: src/lib/date.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  return isDate(date) && isLastDayOfMonth(date);
+}
+
+/**
+ * Calculate the difference in days between two dates
```
[the remark this answers is not in the document]
Comment (drebrez) [reply src/lib/date.ts:22]: mentioned in the comment 👍🏼

File: src/lib/date.ts (hunk 1 above)
Comment (drebrez) [reply src/lib/date.ts]: Renamed to match the name of the date-fns function to be more precise 👍🏼

**Review by neotrow — APPROVED** [review]
thx!

**Pull request merged**